crates/re_types_builder/src/arrow_registry.rs

```diff
@@ -90,23 +90,33 @@ impl ArrowRegistry {
                 .is_some();
             LazyDatatype::Extension(
                 obj.fqname.clone(),
-                Box::new(LazyDatatype::Union(
-                    obj.fields
-                        .iter_mut()
-                        .map(|field| LazyField {
+                Box::new(
+                    // NOTE: Inject the null markers' field first and foremost! That way it is
+                    // guaranteed to be stable and forward-compatible.
+                    LazyDatatype::Union(
+                        std::iter::once(LazyField {
+                            name: "_null_markers".into(),
+                            datatype: LazyDatatype::Null,
+                            // NOTE: The spec doesn't allow a `Null` array to be non-nullable. Not that
+                            // we care either way.
+                            is_nullable: true,
```

Comment:
bit confused by this bit: We use this data type to represent null but the field itself is technically still not null? 🤔

Reply:
TL;DR the `Null` type plays by its own rule (it serializes as a validity bitmap in itself iirc, or straight up nothing if all values are null), so the value you here is irrelevant, but some implementations (e.g. the python one) will complain if you don't set `nullable` to `true`.